### PR TITLE
fix regression errors for all test cases

### PR DESCRIPTION
**Pull request by deepakrkris**
regression errors after checking in request-response-sample
test case fixes

### CODE REVIEW

File: test/client-test.js [hunk 1]
```diff
@@ -442,13 +442,16 @@ describe('SOAP Client', function() {
         var data = {
           Request: {
             $attributes: {
-              $xsiType: '{xmlnsTy}Ty'
+              $xsiType: {
```
Comment (deepakrkris) [inline test/client-test.js]: @raymondfeng it was not supported before, I can add support if you decide

File: src/parser/xmlHandler.js [hunk 2]
```diff
@@ -138,7 +138,7 @@ class XMLHandler {
           nsContext.popContext();
         }
         return node;
-      } else {
+      } else if (typeof val !== "undefined" && typeof val !== null){
         let attrs = val[this.options.attributesKey];
```
Comment (deepakrkris) [inline src/parser/xmlHandler.js]: @raymondfeng  is val==null same as val!==null ??

File: test/client-test.js [hunk 3]
```diff
@@ -442,13 +442,16 @@ describe('SOAP Client', function() {
...
+                "type": "Ty",
+                "xmlns": "xmlnsTy"
```
Comment (deepakrkris) [inline test/client-test.js]: @raymondfeng I will check for consistency with other features and get back to you

File: src/parser/xmlHandler.js (hunk 2 above)
Comment (raymondfeng) [inline src/parser/xmlHandler.js]: Sorry I had a typo. Fixed now

File: test/client-test.js (hunk 1 above)
Comment (raymondfeng) [inline test/client-test.js]: It was not supported by `soap` module but I added it. If it's not working, we need to fix it.

File: src/parser/xmlHandler.js [hunk 4]
```diff
@@ -243,8 +243,13 @@ class XMLHandler {
         let child = attrs[p];
         // if field is $xsiType add xsi:type attribute
         if (p === this.options.xsiTypeKey) {
-          // $xsiType has two fields - type, xmlns
-          let xsiType = QName.parse(child.type, child.xmlns);
+          let xsiType;
+          if(typeof child === "object" && typeof child.type !== "undefined") {
```
Comment (raymondfeng) [inline src/parser/xmlHandler.js]: Please note we use single quote for strings.

**Pull request merged**